naps/platform/beta_platform.py

```diff
@@ -22,6 +22,36 @@ def connect_mainboard(self):
             lvds=["68 70", "74 76", "82 84", "92 94", "93 91", "89 87"]
         )
 
+        # these pins are split between the two connectors so we refer to them directly by FPGA site name
+        lvds_p = "H15 K16 J20 G19 F19 G17 J18 K17 L16 K19 M17 M19 L19 B19 F16 D19 V16 V20 H16 V17 R16 Y18 T20 N18 U14 T16 U18 V15 W14 T11 Y16 T14"
+        lvds_n = "G15 J16 H20 G20 F20 G18 H18 K18 L17 J19 M18 M20 L20 A20 F17 D20 W16 W20 H17 V18 R17 Y19 U20 P19 U15 U17 U19 W15 Y14 T10 Y17 T15"
+
+        lvds_signals = []
+        for i, pn in enumerate(zip(lvds_p.split(" "), lvds_n.split(" "))):
+            lvds_signals.append(Subsignal(f"lvds_{i}",
+                DiffPairs(*pn, dir='i', invert=(i == 18)), # lane 18 is inverted for easier routing
+                Attrs(IOSTANDARD="LVDS_25", DIFF_TERM="TRUE", IBUF_LOW_PWR="TRUE")))
+
+        self.add_resources([
+            Resource("sensor", 0,
+                     Subsignal("clk", Pins("19", dir='o', conn=("JX1", 0)), Attrs(IOSTANDARD="LVCMOS25")),
+                     Subsignal("reset", PinsN("17", dir='o', conn=("JX1", 0)), Attrs(IOSTANDARD="LVCMOS25")),
+                     Subsignal("frame_req", Pins("9", dir='o', conn=("JX1", 0)), Attrs(IOSTANDARD="LVCMOS25")),
+                     Subsignal("t_exp1", Pins("10", dir='o', conn=("JX1", 0)), Attrs(IOSTANDARD="LVCMOS25")),
+                     Subsignal("t_exp2", Pins("100", dir='o', conn=("JX2", 0)), Attrs(IOSTANDARD="LVCMOS25")),
+                     Subsignal("lvds_clk", DiffPairs("81", "83", dir='o', conn=("JX1", 0)), Attrs(IOSTANDARD="LVDS_25", SLEW="SLOW")),
+                     *lvds_signals,
+                     Subsignal("lvds_ctrl", DiffPairs("82", "84", dir='i', conn=("JX2", 0)), Attrs(IOSTANDARD="LVDS_25", DIFF_TERM="TRUE", IBUF_LOW_PWR="TRUE")),
+                     Subsignal("lvds_outclk", DiffPairsN("48", "50", dir='i', conn=("JX2", 0)), Attrs(IOSTANDARD="LVDS_25", DIFF_TERM="TRUE", IBUF_LOW_PWR="TRUE")),
+                     ),
+            Resource("sensor_spi", 0,
```

Comment:
Any specific reason for not using `SPIResource`?

Reply:
SPIResource does not support active high chip select at this time.